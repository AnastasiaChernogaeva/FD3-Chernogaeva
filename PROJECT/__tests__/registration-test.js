import React from 'react';
import renderer from 'react-test-renderer';
 
import Registration from '../components/registration';


test('работа Registration', () => {
 
  // создаём тестовую версию компонента
  const component = renderer.create(
    <Registration />
  );
 
  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
 
  // найдём в вёрстке компонента сами кнопки
 
  const buttonEleme1 = component.root.find( el =>(el.props.id=="FirstR")); 
  const buttonEleme2 = component.root.find( el =>(el.props.id=="SecR")); 
 
  // const buttonEleme1 = component.root.find( el =>(el.props.testPrR1=="buttOn")); 
  // const buttonEleme2 = component.root.find( el =>(el.props.testPrR2=="buttOn"));  

  const buttonElems=[buttonEleme1, buttonEleme2,];
  // находим все кнопки
 
  buttonElems.forEach(elem=>{elem.props.onClick(); // по очереди нажимаем на кнопки и делаем снимк(так два раза)
  
  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  // и "нажмём" на неё
 
  elem.props.onClick();
  
  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
 
})
})  