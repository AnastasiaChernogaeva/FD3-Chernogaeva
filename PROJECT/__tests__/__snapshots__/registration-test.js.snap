// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Registration 1`] = `
<div>
  <div />
  <form
    className="register"
    onChange={[Function]}
  >
    <h1>
      Регистрация
    </h1>
    <label
      htmlFor="NameId"
    >
      Имя
    </label>
    <br />
    <br />
    <input
      id="NameId"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Валентина"
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="LastNameId"
    >
      Фамилия
    </label>
    <br />
    <br />
    <input
      id="LastNameId"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Домаринская"
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <p>
      <b>
        При создании пароля и почты нельзя использовать следующие символы:"!№;%:?*()_-+=".
      </b>
    </p>
    <label
      htmlFor="MailId"
    >
      Электронная почта
    </label>
    <br />
    <br />
    <input
      id="MailId"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="Password"
    >
      Пароль
    </label>
    <br />
    <br />
    <input
      id="Password"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="RePassword"
    >
      Подтвердите пароль
    </label>
    <br />
    <br />
    <input
      id="RePassword"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <h3>
      Дополнительная информация для восстановления доступа к аккаунту
    </h3>
    <label
      htmlFor="pet"
    >
      Введите имя первого домашнего питомца
    </label>
    <br />
    <br />
    <input
      id="pet"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
     
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="color"
    >
      Введите ваш любимый цвет
    </label>
    <br />
    <br />
    <input
      id="color"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
     
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
  </form>
  <div
    className="Top_Buttons"
  >
    <input
      disabled="disabled"
      id="FirstR"
      onClick={[Function]}
      type="button"
      value="Зарегистрироваться"
    />
    <input
      id="SecR"
      onClick={[Function]}
      type="button"
      value="Сбросить все"
    />
  </div>
</div>
`;

exports[`работа Registration 2`] = `
<div>
  <div />
  <form
    className="register"
    onChange={[Function]}
  >
    <h1>
      Регистрация
    </h1>
    <label
      htmlFor="NameId"
    >
      Имя
    </label>
    <br />
    <br />
    <input
      id="NameId"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Валентина"
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="LastNameId"
    >
      Фамилия
    </label>
    <br />
    <br />
    <input
      id="LastNameId"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Домаринская"
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <p>
      <b>
        При создании пароля и почты нельзя использовать следующие символы:"!№;%:?*()_-+=".
      </b>
    </p>
    <label
      htmlFor="MailId"
    >
      Электронная почта
    </label>
    <br />
    <br />
    <input
      id="MailId"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="Password"
    >
      Пароль
    </label>
    <br />
    <br />
    <input
      id="Password"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="RePassword"
    >
      Подтвердите пароль
    </label>
    <br />
    <br />
    <input
      id="RePassword"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <h3>
      Дополнительная информация для восстановления доступа к аккаунту
    </h3>
    <label
      htmlFor="pet"
    >
      Введите имя первого домашнего питомца
    </label>
    <br />
    <br />
    <input
      id="pet"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
     
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="color"
    >
      Введите ваш любимый цвет
    </label>
    <br />
    <br />
    <input
      id="color"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
     
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
  </form>
  <div
    className="Top_Buttons"
  >
    <input
      disabled="disabled"
      id="FirstR"
      onClick={[Function]}
      type="button"
      value="Зарегистрироваться"
    />
    <input
      id="SecR"
      onClick={[Function]}
      type="button"
      value="Сбросить все"
    />
  </div>
</div>
`;

exports[`работа Registration 3`] = `
<div>
  <div />
  <form
    className="register"
    onChange={[Function]}
  >
    <h1>
      Регистрация
    </h1>
    <label
      htmlFor="NameId"
    >
      Имя
    </label>
    <br />
    <br />
    <input
      id="NameId"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Валентина"
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="LastNameId"
    >
      Фамилия
    </label>
    <br />
    <br />
    <input
      id="LastNameId"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Домаринская"
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <p>
      <b>
        При создании пароля и почты нельзя использовать следующие символы:"!№;%:?*()_-+=".
      </b>
    </p>
    <label
      htmlFor="MailId"
    >
      Электронная почта
    </label>
    <br />
    <br />
    <input
      id="MailId"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="Password"
    >
      Пароль
    </label>
    <br />
    <br />
    <input
      id="Password"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="RePassword"
    >
      Подтвердите пароль
    </label>
    <br />
    <br />
    <input
      id="RePassword"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <h3>
      Дополнительная информация для восстановления доступа к аккаунту
    </h3>
    <label
      htmlFor="pet"
    >
      Введите имя первого домашнего питомца
    </label>
    <br />
    <br />
    <input
      id="pet"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
     
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="color"
    >
      Введите ваш любимый цвет
    </label>
    <br />
    <br />
    <input
      id="color"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
     
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
  </form>
  <div
    className="Top_Buttons"
  >
    <input
      disabled="disabled"
      id="FirstR"
      onClick={[Function]}
      type="button"
      value="Зарегистрироваться"
    />
    <input
      id="SecR"
      onClick={[Function]}
      type="button"
      value="Сбросить все"
    />
  </div>
</div>
`;

exports[`работа Registration 4`] = `
<div>
  <div />
  <form
    className="register"
    onChange={[Function]}
  >
    <h1>
      Регистрация
    </h1>
    <label
      htmlFor="NameId"
    >
      Имя
    </label>
    <br />
    <br />
    <input
      id="NameId"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Валентина"
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="LastNameId"
    >
      Фамилия
    </label>
    <br />
    <br />
    <input
      id="LastNameId"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Домаринская"
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <p>
      <b>
        При создании пароля и почты нельзя использовать следующие символы:"!№;%:?*()_-+=".
      </b>
    </p>
    <label
      htmlFor="MailId"
    >
      Электронная почта
    </label>
    <br />
    <br />
    <input
      id="MailId"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="Password"
    >
      Пароль
    </label>
    <br />
    <br />
    <input
      id="Password"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="RePassword"
    >
      Подтвердите пароль
    </label>
    <br />
    <br />
    <input
      id="RePassword"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <h3>
      Дополнительная информация для восстановления доступа к аккаунту
    </h3>
    <label
      htmlFor="pet"
    >
      Введите имя первого домашнего питомца
    </label>
    <br />
    <br />
    <input
      id="pet"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
     
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="color"
    >
      Введите ваш любимый цвет
    </label>
    <br />
    <br />
    <input
      id="color"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
     
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
  </form>
  <div
    className="Top_Buttons"
  >
    <input
      disabled="disabled"
      id="FirstR"
      onClick={[Function]}
      type="button"
      value="Зарегистрироваться"
    />
    <input
      id="SecR"
      onClick={[Function]}
      type="button"
      value="Сбросить все"
    />
  </div>
</div>
`;

exports[`работа Registration 5`] = `
<div>
  <div />
  <form
    className="register"
    onChange={[Function]}
  >
    <h1>
      Регистрация
    </h1>
    <label
      htmlFor="NameId"
    >
      Имя
    </label>
    <br />
    <br />
    <input
      id="NameId"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Валентина"
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="LastNameId"
    >
      Фамилия
    </label>
    <br />
    <br />
    <input
      id="LastNameId"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Домаринская"
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <p>
      <b>
        При создании пароля и почты нельзя использовать следующие символы:"!№;%:?*()_-+=".
      </b>
    </p>
    <label
      htmlFor="MailId"
    >
      Электронная почта
    </label>
    <br />
    <br />
    <input
      id="MailId"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="Password"
    >
      Пароль
    </label>
    <br />
    <br />
    <input
      id="Password"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="RePassword"
    >
      Подтвердите пароль
    </label>
    <br />
    <br />
    <input
      id="RePassword"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <h3>
      Дополнительная информация для восстановления доступа к аккаунту
    </h3>
    <label
      htmlFor="pet"
    >
      Введите имя первого домашнего питомца
    </label>
    <br />
    <br />
    <input
      id="pet"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
     
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
    <label
      htmlFor="color"
    >
      Введите ваш любимый цвет
    </label>
    <br />
    <br />
    <input
      id="color"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="text"
      value=""
    />
     
    <span
      className="error"
    >
      
    </span>
    <br />
    <br />
  </form>
  <div
    className="Top_Buttons"
  >
    <input
      disabled="disabled"
      id="FirstR"
      onClick={[Function]}
      type="button"
      value="Зарегистрироваться"
    />
    <input
      id="SecR"
      onClick={[Function]}
      type="button"
      value="Сбросить все"
    />
  </div>
</div>
`;
